### PR TITLE
[DevPortal][Application Management]Add SAML configuration to applications 

### PR DESCRIPTION
**Pull request by inthirakumaaran**
## Goals
- Add SAML configuration to applications
image

- Add form to capture SAML configuration in creation journey

### CODE REVIEW

File: apps/admin-portal/src/models/application-inbound.ts [hunk 1]
```diff
@@ -160,3 +162,109 @@ export enum SupportedAuthProtocolMetaTypes {
...
+    enableIdpInitiatedSingleSignOn?: boolean;
+    assertion?: SAMLAssertionConfigurationInterface;
+}
+
+export interface SAMLAttributeProfileInterface {
+    enabled?: boolean;
+    alwaysIncludeAttributesInResponse?: boolean;
+}
+
+export enum LogoutMethod {
```
Comment (brionmario) [inline apps/admin-portal/src/models/application-inbound.ts]: Would be better if this is renamed to `LogoutMethods`

File: apps/admin-portal/src/models/global.ts [hunk 2]
```diff
@@ -25,3 +25,7 @@ export interface GlobalInterface {
     isApplicationsPageVisible: boolean;
     isGlobalLoaderVisible: boolean;
 }
+
+export type MandatoryArray<T> = {
```
Comment (brionmario) [inline apps/admin-portal/src/models/global.ts]: Shall we add a doc comment to explain what this does?

File: apps/admin-portal/src/components/applications/forms/inbound-saml-form.tsx [hunk 3]
```diff
@@ -0,0 +1,789 @@
...
+                    signingAlgorithm: values.get("signingAlgorithm")
+                },
+                enableAssertionQueryProfile: values.get("assertionQueryProfile").includes("enableAssertionQueryProfile")
+            }
+        }
+    };
+
+
+    return (
+        <>
```
Comment (brionmario) [inline apps/admin-portal/src/components/applications/forms/inbound-saml-form.tsx]: We can further simplify as follows. Wrapping using a fragment and curly braces is not required here.

```tsx
return (
    metadata && (
           ...
    )
)
```

File: apps/admin-portal/src/components/applications/forms/inbound-saml-form.tsx [hunk 4]
```diff
@@ -0,0 +1,785 @@
...
+                },
+                responseSigning: {
+                    enabled: values.get("responseSigning").includes("enabled"),
+                    signingAlgorithm: values.get("signingAlgorithm")
+                },
+                enableAssertionQueryProfile: values.get("assertionQueryProfile").includes("enableAssertionQueryProfile")
+            }
+        }
+    };
+
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/applications/forms/inbound-saml-form.tsx]:
```suggestion
```

File: apps/admin-portal/src/components/applications/wizard/saml-protocol-settings-wizard-form.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+        triggerSubmit,
+        onSubmit
+    } = props;
+
+
+    const [assertionConsumerUrls, setAssertionConsumerUrls] = useState("");
+    const [showAssertionConsumerUrlError, setAssertionConsumerUrlError] = useState(false);
+
+    const form = useRef(null);
+
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/applications/wizard/saml-protocol-settings-wizard-form.tsx]:
```suggestion
```

File: apps/admin-portal/src/components/applications/wizard/saml-protocol-settings-wizard-form.tsx [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+export const SAMLProtocolSettingsWizardForm: FunctionComponent<SAMLProtocolSettingsWizardFormPropsInterface> = (
+    props: SAMLProtocolSettingsWizardFormPropsInterface
+): ReactElement => {
+
+    const {
+        initialValues,
+        triggerSubmit,
+        onSubmit
+    } = props;
+
```
Comment (jeradrutnam) [inline apps/admin-portal/src/components/applications/wizard/saml-protocol-settings-wizard-form.tsx]:
```suggestion
```

**Pull request merged**